python-checks/src/main/java/org/sonar/python/checks/BooleanCheckNotInvertedCheck.java

```diff
@@ -120,8 +122,33 @@ private static void createQuickFix(IssueWithQuickFix issue, String oppositeOpera
   }
 
   private static PythonTextEdit getReplaceEdit(BinaryExpression toUse, Expression toReplace, String oppositeOperator) {
-    List<Tree> children = toReplace.parent().parent().children();
+    // To be able to remove the 'not' sign and the possible space afterwards
+    Tree notAncestor = Optional.ofNullable(TreeUtils.firstAncestorOfKind(toReplace, Kind.NOT))
+      .orElseThrow(() -> new IllegalArgumentException("A NOT tree must be found. It was not the case."));
+
+    List<Tree> children = notAncestor.children();
     return PythonTextEdit.replaceRange(children.get(0), children.get(children.size() - 1),
-      toUse.leftOperand().firstToken().value() + " " + oppositeOperator + " " + toUse.rightOperand().firstToken().value());
+      getNewExpression(toUse, oppositeOperator));
+  }
+
+  private static String getNewExpression(BinaryExpression toUse, String oppositeOperator) {
+    return getText(toUse.leftOperand()) + " " + oppositeOperator + " " + getText(toUse.rightOperand());
+  }
+
+  private static String getText(Tree tree) {
+    if (tree.is(Kind.TOKEN)) {
+      return ((Token) tree).value();
+    }
+
+    return tree.children().stream()
+      .map(BooleanCheckNotInvertedCheck::getText)
+      .reduce("", (acc, currentChar) -> isSpaceNotNeeded(acc, currentChar) ? (acc + currentChar) : (acc + " " + currentChar));
```

Comment:
I think there is still a problem with "f strings" like the following:
```java
    codeWithIssue = "x = not (a == f'foo${b}')";
    codeFixed = "x = a != f'foo${b}'";
    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
```

I would suggest to use `TreeUtils.tokens(tree)` which handles already f-strings:
```suggestion
    return TreeUtils.tokens(tree).stream()
      .map(Token::value)
      .reduce("", (acc, currentChar) -> isSpaceNotNeeded(acc, currentChar) ? (acc + currentChar) : (acc + " " + currentChar));
```

Reply:
Yes, I didn't cover this. It should be the case now